### PR TITLE
whatsnew 1.3.0

### PR DESCRIPTION
**Pull request by rhshadrach**
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

Minor fixups. After I get feedback here, I plan to make a PR with detailed instructions about adding to the whatsnew.

cc @attack68 - the Styler sections had a major revisions, wanted to get any feedback here.

cc @jorisvandenbossche @simonjayhawkins

### CODE REVIEW

File: doc/source/whatsnew/v1.3.0.rst
Comment (jreback) [inline doc/source/whatsnew/v1.3.0.rst:143]: can use bullet points to improve readability

File: doc/source/whatsnew/v1.3.0.rst
Comment (jreback) [inline doc/source/whatsnew/v1.3.0.rst:208]: might be worth it to:
- group the other enhancements by category / type (groupby / rolling / styler etc). and create-sub headers

File: doc/source/whatsnew/v1.3.0.rst
Comment (jreback) [inline doc/source/whatsnew/v1.3.0.rst:935]: I *think* we have a MultiIndex section? if so can you move relevant notes

File: doc/source/whatsnew/v1.3.0.rst
Comment (jreback) [inline doc/source/whatsnew/v1.3.0.rst:973]: ideally can group notes e.g. for read_csv, read_excel etc. sub-sections prob not needed)

File: doc/source/whatsnew/v1.3.0.rst
Comment (jreback) [inline doc/source/whatsnew/v1.3.0.rst:1114]: any of these that can be in other more specific sections would be good.

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v1.3.0.rst
Comment (rhshadrach) [inline doc/source/whatsnew/v1.3.0.rst:644]: I've removed this note as I think it only pertains to the internals - though I understand downstream packages are using it. Perhaps this is a case where we'd like it in the whatsnew - cc @topper-123

File: doc/source/whatsnew/v1.3.0.rst
Comment (rhshadrach) [reply doc/source/whatsnew/v1.3.0.rst]: Agreed - while reworking this as bullet points I noticed that much is a duplicate of other lines in the whatsnew. We should either have a single section in Enhancements or single lines in the Other Enhancements (which will be broken into sections, and hence include a Styler section) and Bugfixes.

I lean toward breaking it up (since they aren't all enhancements) into the appropriate sections. cc @attack68 for thoughts.

Comment (rhshadrach) [reply doc/source/whatsnew/v1.3.0.rst]: Marking this PR as a draft until this is resolved so it won't be merged.

**Comment by rhshadrach** [comment]
@jreback - agreed with all your suggestions above but would like to hold off and do in a followup to keep the diff here reasonable. I think it's okay for the changes to happen during the RC period (but won't be waiting for the RC period either)

**Comment by rhshadrach** [comment]
@attack68 - The performance improvement of the HTML rendering referenced #37792 but this is unrelated, not sure what issue/PR was meant. Also from the line

> - The method :meth:`.Styler.format` has been improved to easily format missing data, precision, and perform HTML escaping (:issue:`40437`, :issue:`40134`)

I think the "format missing data" is referring to #40134, but I can't find changes to na_rep there. Can you help me understand what changed?

File: doc/source/whatsnew/v1.3.0.rst
Comment (topper-123) [reply doc/source/whatsnew/v1.3.0.rst:644]: I consider this public for developers, who subclass dataframes and series. See [here](https://pandas.pydata.org/docs/development/extending.html#subclassing-pandas-data-structures), where the attributes were (publically) explained.

However this already raised previously, and this PR just changes the exception type, so it's probably not that important. I'd prefer to keep it, but am ok with deleting.

**Review by attack68 — APPROVED** [review]
lgtm.

after many commits it became a bit messy. the bullet point format seems better. Also there were some Styler additions in relevant sections below this one, so some things might have been duplicated, but maybe thats ok for readers?

**Comment by attack68** [comment]
> @attack68 - The performance improvement of the HTML rendering referenced #37792 but this is unrelated, not sure what issue/PR was meant. Also from the line
> 
> > * The method :meth:`.Styler.format` has been improved to easily format missing data, precision, and perform HTML escaping (:issue:`40437`, :issue:`40134`)
> 
> I think the "format missing data" is referring to #40134, but I can't find changes to na_rep there. Can you help me understand what changed?

there was a function called `Styler.set_na_rep()`, which set a property on `self` and the `format` method with an `na_rep` arg which conflicted at times. So, yes, it was there previously but it didn't always do what you thought. Now the missing data in `Styler` is handled exclusively and consistently by `Styler`, and `set_na_rep` is deprecated.

**Comment by attack68** [comment]
> @attack68 - The performance improvement of the HTML rendering referenced #37792 but this is unrelated, not sure what issue/PR was meant. Also from the line

#39972 and #39952

File: doc/source/whatsnew/v1.3.0.rst
Comment (jbrockmendel) [reply doc/source/whatsnew/v1.3.0.rst:928]: i think this belongs here in the Indexing section, since the user-facing bug is in `.loc`

File: doc/source/whatsnew/v1.3.0.rst
Comment (jbrockmendel) [inline doc/source/whatsnew/v1.3.0.rst:689]: Not necessarily for this PR, but there are enough deprecations i think it might be worth subsections for e.g. positional arguments (or at least collect them all together)

**Comment by jbrockmendel** [comment]
@rhshadrach there's a lot here.  id suggest breaking off the non-controversial bits into a precursor PR

**Comment by simonjayhawkins** [comment]
> @rhshadrach there's a lot here. id suggest breaking off the non-controversial bits into a precursor PR

maybe start with the sphinx directive errors, then another PR for the sphinx directives with broken links, then the grammer and then finally move stuff.

**Comment by simonjayhawkins** [comment]
The first one (maybe 2) would be good for the rc

**Comment by rhshadrach** [comment]
Going back on what I said in https://github.com/pandas-dev/pandas/pull/41747#issuecomment-851779839, I only plan to finish up this what's in this PR. I'll break it up as @simonjayhawkins recommended.

File: doc/source/whatsnew/v1.3.0.rst
Comment (jbrockmendel) [inline doc/source/whatsnew/v1.3.0.rst:1101]: "caused" -> "causing"?

File: doc/source/whatsnew/v1.3.0.rst
Comment (jbrockmendel) [inline doc/source/whatsnew/v1.3.0.rst:1086]: why removed `:class:` from Series? no real objection, just not clear on if i should change my habits when writing these

Comment (jbrockmendel) [inline doc/source/whatsnew/v1.3.0.rst:1086]: ExtensionArray dtype -> ExtensionDtype?

File: doc/source/whatsnew/v1.3.0.rst
Comment (jorisvandenbossche) [reply doc/source/whatsnew/v1.3.0.rst:644]: It's correct that `_constructor_expanddim` is "public" developer API.

File: doc/source/whatsnew/v1.3.0.rst
Comment (rhshadrach) [reply doc/source/whatsnew/v1.3.0.rst:1086]: This originated from https://github.com/pandas-dev/pandas/pull/38002#discussion_r528531153 and I agree with @jorisvandenbossche's assessment - links on Series and DataFrame serve only as a distraction in cases such as these. I think a good general rule is to only link when Series/DataFrame are the main subject of the change.

**Comment by jbrockmendel** [comment]
@rhshadrach is more of this getting broken off or can you mark as Ready For Review?

**Comment by rhshadrach** [comment]
@jbrockmendel - one last one (#41822).

The main reason for draft status is https://github.com/pandas-dev/pandas/pull/41747#discussion_r642758379. Currently the top section of styler duplicates many of the other notes in the whatsnew. Need to deduplicate and decide whether enhancements for the Styler should appear in the top enhancement section or as line items in the Other Enhancements (grouped together). Any thoughts here would be appreciated.

I think it is clear that bugfixes should go in the Styler bugfixes section and not in the top enhancement section.

**Comment by rhshadrach** [comment]
I've deduplicated the Styler notes and reorganized so that all enhancements are in `Enhancements` in their own section, while Performance, Deprecation, and Bugfix notes appear below.

File: doc/source/whatsnew/v1.3.0.rst
Comment (rhshadrach) [inline doc/source/whatsnew/v1.3.0.rst:125]: @attack68 - I don't see where a raise was added in #39660. Am I just missing it?

I'd like to verify that apply nor applymap will themselves raise, but rather after using them when the Styler renders is were it raises.

File: doc/source/whatsnew/v1.3.0.rst
Comment (rhshadrach) [reply doc/source/whatsnew/v1.3.0.rst:645]: Thanks @topper-123 and @jorisvandenbossche - this note has been restored.

**Comment by rhshadrach** [comment]
I can siphon off the Styler relevant pieces into a separate PR as well if that will make this easier to review. Let me know if that's desirable @simonjayhawkins, @jbrockmendel  (or anyone else)

File: doc/source/whatsnew/v1.3.0.rst
Comment (attack68) [reply doc/source/whatsnew/v1.3.0.rst:125]: - `Styler.apply/applymap` necessarily calls `_update_ctx`, **upon render.**
 - `_update_ctx` was changed to call `_maybe_convert_css_to_tuples`. 
 - `_maybe_convert_css_to_tuples` raises on bad CSS.

When `apply/applymap` are called all they do is register themselves in a TODO array to be actually performed later.

File: doc/source/whatsnew/v1.3.0.rst
Comment (attack68) [inline doc/source/whatsnew/v1.3.0.rst:126]: and LaTeX escaping as of #41619

File: doc/source/whatsnew/v1.3.0.rst
Comment (attack68) [inline doc/source/whatsnew/v1.3.0.rst:132]: this is duplicated with line 123

File: doc/source/whatsnew/v1.3.0.rst
Comment (attack68) [inline doc/source/whatsnew/v1.3.0.rst:133]: duplicated with line 126

File: doc/source/whatsnew/v1.3.0.rst
Comment (attack68) [inline doc/source/whatsnew/v1.3.0.rst:138]: spelling of styler

Comment (attack68) [reply doc/source/whatsnew/v1.3.0.rst:138]: and it was renamed by jreback to `styler.render.max_elements`

**Review by attack68 — CHANGES_REQUESTED** [review]
see a few inline comments

File: doc/source/whatsnew/v1.3.0.rst
Comment (simonjayhawkins) [inline doc/source/whatsnew/v1.3.0.rst:166]: I think we need a `.. _whatsnew_130.centered_datetimelike_rolling_window:` here

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]
I would say merge this now. and can do another pass to address any outstanding comments, but web and docs ci is failing. can you fix that up.

**Comment by simonjayhawkins** [comment]
@rhshadrach merge conflicts

File: doc/source/whatsnew/v1.3.0.rst
Comment (rhshadrach) [reply doc/source/whatsnew/v1.3.0.rst:125]: Thanks!

**Comment by rhshadrach** [comment]
Thanks @attack68 and @simonjayhawkins for the comments/requests, adjustments made and conflicts resolved.

File: doc/source/whatsnew/v1.3.0.rst
Comment (simonjayhawkins) [reply doc/source/whatsnew/v1.3.0.rst]: @MarcoGorelli is planning to tidy that up. https://github.com/pandas-dev/pandas/issues/41485#issuecomment-854714232

**Comment by simonjayhawkins** [comment]
Thanks @rhshadrach I think just @jreback comments about moving stuff to do in follow-up.

File: doc/source/whatsnew/v1.3.0.rst
Comment (MarcoGorelli) [reply doc/source/whatsnew/v1.3.0.rst]: yup - those deprecations are all done now so I'll do this week

**Pull request merged**
